Source/SaveStates/SaveState.cs

```diff
@@ -337,19 +361,96 @@ private IEnumerator LoadStateCoro(bool loadDuped)
             ReflectionHelper.CallMethod(HeroController.instance, "FinishedEnteringScene", true, false);
             ReflectionHelper.CallMethod(GameManager.instance, "UpdateUIStateFromGameState");
 
-            if (data.roomSpecificOptions != "0")
+            if (data.roomSpecificOptions != "0" && data.roomSpecificOptions != null)
             {
                 RoomSpecific.DoRoomSpecific(data.saveScene, data.roomSpecificOptions);
             }
 
-            yield return null;
+            //moving this here seems to work now? no need to toggle canvas
+            //preserve correct hp amount
+            bool isInfiniteHp = DebugMod.infiniteHP;
+            DebugMod.infiniteHP = false;
+            //prevent flower break by taking it, then giving it back
+            PlayerData.instance.hasXunFlower = false;
+            PlayerData.instance.health = data.savedPd.health;
+            HeroController.instance.TakeHealth(1);
```

Comment:
Since the main purpose is to fix UI can we not do this? We can just enable and disable the health parent

Reply:
Main issue is HUD just takes forever to build